### PR TITLE
Feature/multivalued

### PR DESCRIPTION
**Pull request by mtnbarreto**
Multivalued sections feature added..

### CODE REVIEW

**Comment by mtnbarreto** [comment]
@m-revetria @mats-claassen Could you please review this pr?

File: README.md [hunk 1]
```diff
@@ -421,6 +440,50 @@ Additionally you can setup list of options to be grouped by sections using follo
...
 
+### Multivalued Sections
+
+Eureka supports multiple values for a certain field (such as telephone numbers in a contact) by using Multivalued sections. It allows us to easily create insertable, deletable and reorderable sections.
+
+<img src="Example/Media/EurekaMultivalued.gif" width="300" alt="Screenshot of Multivalued Section" />
+
+#### How to create a multivalued section
+
+In order to create a multivalued section we have to use `MultivaluedSection` type instead of the regular `Section` type. `MultivaluedSection` extends `Section` type and has some additional properties to configure multivalued section behavior.
```
Comment (mats-claassen) [inline README.md]: `MultivaluedSection` extends `Section` ~type~

File: Source/Core/Core.swift [hunk 2]
```diff
@@ -740,6 +756,126 @@ extension FormViewController : UITableViewDelegate {
...
+    
+    public func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let row = form[indexPath]
+            var section = row.section!
+            if let _ = row.baseCell.findFirstResponder() {
+                tableView.endEditing(true)
+            }
+            section.remove(at: indexPath.row)
+            DispatchQueue.main.asyncAfter(deadline: .now()) {
```
Comment (mats-claassen) [inline Source/Core/Core.swift]: Why not `DispatchQueue.main.async`

File: Source/Core/Core.swift [hunk 3]
```diff
@@ -740,6 +756,126 @@ extension FormViewController : UITableViewDelegate {
...
+            }
+        }
+        else if editingStyle == .insert {
+            guard var section = form[indexPath.section] as? MultivaluedSection else { return }
+            guard let multivaluedRowToInsertAt = section.multivaluedRowToInsertAt else {
+                fatalError("Multivalued section multivaluedRowToInsertAt property must be set up")
+            }
+            let newRow = multivaluedRowToInsertAt(max(0, section.count - 1))
+            section.insert(newRow, at: section.count - 1)
+            DispatchQueue.main.asyncAfter(deadline: .now()) {
```
Comment (mats-claassen) [inline Source/Core/Core.swift]: the same here

File: Source/Core/Section.swift [hunk 4]
```diff
@@ -374,3 +374,71 @@ extension Section /* Condition */{
...
+    public static let Insert = MultivaluedOptions(.insert)
+    
+    /// Allows user to delete rows.
+    public static let Delete = MultivaluedOptions(.delete)
+    
+    /// Allows user to reorder rows
+    public static let Reorder = MultivaluedOptions(.reorder)
+}
+
+open class MultivaluedSection: Section {
```
Comment (mats-claassen) [inline Source/Core/Section.swift]: Missing code documentation

File: Source/Core/Core.swift (hunk 2 above)
Comment (mtnbarreto) [inline Source/Core/Core.swift]: Originally i was adding some delay ( DispatchQueue.main.asyncAfter(deadline: .now() + nanoseconds(xyz))) but then i realized there were no need for that block execution dalay.

If it is exactly the same i would also like to use `DispatchQueue.main.async`.

**Pull request merged**